OminosOminos/ominoes.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Ominoes do
+
+    def omino_game(index,x,r,c) do
+    IO.puts "Before firstIndicator"
+    firstIndicator = 7
+
+    multiplicationResult = r * c
+
+    secondIndicator = rem(multiplicationResult, x)
+
+    thirdIndicator = x
+
+    fourthIndicator = 0
+
+    if rem(x,2) == 0 do
+     fourthIndicator = x / 2
+    else
+     fourthIndicator = (x/2) + 1
+    end
+
+    winner = "Richard"
+
+    if x < firstIndicator do
+     if secondIndicator == 0 do
+       if r >= thirdIndicator || c >= thirdIndicator do
+        if r >= thirdIndicator do
+        if c >= fourthIndicator do
+            winner = "Gabriel"
+        end
+        if c >= thirdIndicator do
+            winner = "Gabriel"
+        end
+        end
+       end
+     end
+    end
+    File.write("output.txt", "Case ##{index}: #{winner}\n", [:append])
+    IO.puts "winner #{winner}"
+    end
+
+end
```

Comment:
Hi, Dayra! Let us get to work on this and we'll get back to you with our comments. Thanks!

Reply:
We got the answer Dayra : D, instead of using a nested if's, we found the `cond` control structure, which helps to use an amount of conditions and set them as a group to set actions for each case. We use the already built function, so we just cleaned the code by erasing the unnecessary variables and focus only to get the winner. I will send the code as a comment here.

Reply:
```defmodule Ominoes do
    def omino_game(index,x,r,c) do
    s = min(r, c)
    l = max(r, c)  
    cond do 
      x >= 7 -> winner = "RICHARD"
      rem(r * c, x) != 0 -> winner = "RICHARD"
      l < x -> winner = "RICHARD"
      s < trunc(x / 2 + 1) and x >= 3 -> winner = "RICHARD"
      x == 5 and (l== 5 and s == 3) -> winner = "RICHARD"
      true -> winner = "GABRIEL"
    end
    File.write("output.txt", "Case ##{index}: #{winner}\r\n", [:append])
    end
end

  File.stream!("D-small-practice.in")
  |> Stream.map(&String.strip/1)
  |> Stream.with_index
  |> Stream.map(fn ({line, index}) ->
    if index != 0 do
      list_of_inputs = String.split(line)
      xx = String.to_integer(Enum.at(list_of_inputs, 0))
      rr = String.to_integer(Enum.at(list_of_inputs, 1))
      cc = String.to_integer(Enum.at(list_of_inputs, 2))
      Ominoes.omino_game(index,xx,rr,cc)
    end
  end)
  |> Stream.run

Reply:
hey, almost there 👍  you also can go through the file twice
Every time you do a map, you go through the file, there is no need to do it twice ... you can do it within the last map and also you can save that in each iteration verify this "if index! = 0 do". 
It does not affect the performance but for that, it's the first line of the file, so you know how many elements you have to take. :)